extension/src/extension.ts

```diff
@@ -218,57 +216,101 @@ async function runCodeNatively(document: vscode.TextDocument) {
 
   const user32Path = getPath(["irvine", "User32.Lib"]);
 
-  //creates a new file based on the document the user is working on
-  const tempFile = document;
+  const currentDirectory = document.uri.fsPath.slice(
+    0,
+    document.uri.fsPath.length - basename(document.fileName).length
+  );
+
+  const baseFilename = basename(document.fileName).slice(
+    0,
+    -extname(document.fileName).length
+  ).replaceAll(/\s/g,"_");
+  const tempBaseFilename =
+    (basename(document.fileName).slice(0, -extname(document.fileName).length) +
+    "-temp").replaceAll(/\s/g,"_");
```

Comment:
why is it "-temp" and not ".temp"

Reply:
It was i was testing something while trying to get gitbash to work i forgot to change it back I will change that to .temp